pkg/tfgen/renames.go

```diff
@@ -102,40 +102,116 @@ func (r *renamesBuilder) registerNamedObjectType(paths paths.TypePathSet, token
 	for _, p := range paths.Paths() {
 		r.objectTypes[p.String()] = token
 	}
+	r.objectTypePaths[token] = paths
 }
 
 func (r *renamesBuilder) registerProperty(parent paths.TypePath, name paths.PropertyName) {
-	if name.Key == string(name.Name) {
-		return
+	r.properties = append(r.properties, struct {
+		parent paths.TypePath
+		name   paths.PropertyName
+	}{parent: parent, name: name})
+}
+
+func (*renamesBuilder) isConfig(propertyParentPath paths.TypePath) bool {
+	_, ok := propertyParentPath.(*paths.ConfigPath)
+	return ok
+}
+
+func (r *renamesBuilder) propetyParentToken(propertyParentPath paths.TypePath) (tokens.Token, error) {
+	switch parent := propertyParentPath.(type) {
+	case *paths.ConfigPath:
+		return "", fmt.Errorf("ConfigPath should have been detected with isConfig")
```

Comment:
I don't have context, but the message implies that this should never happen, regardless of user input. Should we just `panic` here?

Reply:
In theory yes, in practice I'm leery of panics since I had to work through quite a few of those last night. I think the trouble is that I don't fully trust the TypePath steppers yet. In particular the path that has an ElementPath fragment in it - that is going from `List[String]` into `String` must be aware of Pulumi flattening but isn't aware yet, so the code has hacks to compensate for this..

I think I'll leave it as not having any panics now (and clean up existing ones) but we can tighten up once we have more confidence from a large-scale rollout.

Reply:
My concern is that we won't get confidence from a large scale roll out because we have so many warnings, which makes our warnings easy to miss.

Reply:
Yes, ok this one is a panic. I think what you're implicitly asking for also is raising the warnings to errors here, we can come back to this in a few weeks.